examples_tests/20.Megatexture/main.cpp

```diff
@@ -264,47 +264,68 @@ class EventReceiver : public nbl::IEventReceiver
-	//! and we don't want a jittery cursor in the middle distracting us
-	device->getCursorControl()->setVisible(false);
-
-	//! Since our cursor will be enslaved, there will be no way to close the window
-	//! So we listen for the "Q" key being pressed and exit the application
-	EventReceiver receiver;
-	device->setEventReceiver(&receiver);
-
-    auto* driver = device->getVideoDriver();
-    auto* smgr = device->getSceneManager();
-    auto* am = device->getAssetManager();
+    constexpr uint32_t WIN_W = 1280;
+    constexpr uint32_t WIN_H = 720;
+    constexpr uint32_t FBO_COUNT = 1u;
+
+    auto initOutput = CommonAPI::Init<WIN_W, WIN_H, FBO_COUNT>(video::EAT_OPENGL, "MeshLoaders");
+    auto window = std::move(initOutput.window);
+    auto gl = std::move(initOutput.apiConnection);
+    auto surface = std::move(initOutput.surface);
+    auto gpuPhysicalDevice = std::move(initOutput.physicalDevice);
+    auto logicalDevice = std::move(initOutput.logicalDevice);
+    auto queue = std::move(initOutput.queue);
+    auto swapchain = std::move(initOutput.swapchain);
+    auto renderpass = std::move(initOutput.renderpass);
+    auto fbo = std::move(initOutput.fbo[0]);
+    auto commandPool = std::move(initOutput.commandPool);
+
+    core::smart_refctd_ptr<nbl::video::IGPUCommandBuffer> commandBuffers[1];
+    logicalDevice->createCommandBuffers(commandPool.get(), nbl::video::IGPUCommandBuffer::EL_PRIMARY, 1, commandBuffers);
+    auto commandBuffer = commandBuffers[0];
+
+    core::smart_refctd_ptr<nbl::asset::IAssetManager> assetManager;
+    {
+        nbl::core::smart_refctd_ptr<nbl::io::IFileSystem> fileSystem = nbl::core::make_smart_refctd_ptr<nbl::io::CFileSystem>("");
+        assetManager = core::make_smart_refctd_ptr<nbl::asset::IAssetManager>(std::move(fileSystem));
+    }
+
+    nbl::video::IGPUObjectFromAssetConverter cpu2gpu;
+    nbl::video::IGPUObjectFromAssetConverter::SParams cpu2gpuParams;
+    cpu2gpuParams.assetManager = assetManager.get();
+    cpu2gpuParams.device = logicalDevice.get();
+    cpu2gpuParams.finalQueueFamIx = queue->getFamilyIndex();
+    cpu2gpuParams.limits = gpuPhysicalDevice->getLimits();
+    cpu2gpuParams.pipelineCache = nullptr;
+    cpu2gpuParams.sharingMode = nbl::asset::ESM_EXCLUSIVE;
+    cpu2gpuParams.perQueue[nbl::video::IGPUObjectFromAssetConverter::EQU_TRANSFER].queue = queue;
+    cpu2gpuParams.perQueue[nbl::video::IGPUObjectFromAssetConverter::EQU_COMPUTE].queue = queue;
+
+    auto createDescriptorPool = [&](const uint32_t textureCount)
+    {
+        constexpr uint32_t maxItemCount = 256u;
+        {
+            nbl::video::IDescriptorPool::SDescriptorPoolSize poolSize;
+            poolSize.count = textureCount;
+            poolSize.type = nbl::asset::EDT_COMBINED_IMAGE_SAMPLER;
+            return logicalDevice->createDescriptorPool(static_cast<nbl::video::IDescriptorPool::E_CREATE_FLAGS>(0), maxItemCount, 1u, &poolSize);
+        }
```

Comment:
@AnastaZIuk looking at your usage, it looks like you've messed up the semantics here

it should be a pool of UBOs